hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/EmbeddedForeignKeyDescriptor.java

```diff
@@ -330,17 +324,18 @@ public void visitTargetColumns(ColumnConsumer consumer) {
 	}
 
 	@Override
-	public boolean areTargetColumnNamesEqualsTo(String[] columnNames) {
-		int length = columnNames.length;
-		if ( length != targetColumnExpressions.size() ) {
-			return false;
-		}
-		for ( int i = 0; i < length; i++ ) {
-			if ( !targetColumnExpressions.contains( columnNames[i] ) ) {
-				return false;
+	public AssociationKey getAssociationKey() {
+		if ( associationKey == null ) {
+			String[] associationKeyColumns = new String[keyColumnExpressions.size()];
+			int i = 0;
+			for ( String keyColumnExpression : this.keyColumnExpressions ) {
+				associationKeyColumns[i] = keyColumnExpression;
+				i++;
 			}
```

Comment:
I am pretty sure `System.arrayCopy` seems to be much more performant, though not a big deal.